borg/hashindex.pyx

```diff
@@ -63,7 +71,9 @@ cdef class IndexBase:
             path = os.fsencode(path)
             self.index = hashindex_read(path)
             if not self.index:
-                raise Exception('hashindex_read failed')
+                if errno:
+                    raise OSError(errno, strerror(errno).decode(), os.fsdecode(path))
```

Comment:
Hm, I'll need to check whether "strerror(errno).decode()" is correct... probably either locale-dependent encoding or the FS encoding.